Makefile

```diff
@@ -0,0 +1,163 @@
+mkfile_dir := $(dir $(abspath $(lastword $(MAKEFILE_LIST))))
+
+##### Variables used
+# (can be overriden using command line args to make)
+# Paths used in build.
+BUILD_DIR?=$(abspath ${mkfile_dir}/build)
+GECKO_SOURCE_DIR?=$(abspath ${mkfile_dir}/gecko-dev)
+GAIA_SOURCE_DIR?=$(abspath ${mkfile_dir}/gaia)
+# Used to increment .deb package version
+DATETIME?=$(shell date +%d%m%y%H%M)
+GECKO_VERSION?=46
+CUSTOM_MOZCONFIG?=
+RESOLUTION?=800x600
+B2G_PROFILE_PARENT?=${HOME}/.mozilla/b2g
+B2G_PROFILE_NAME?=xsession.profile
+######
+
+##### Git repo validity check
+# gecko/gaia must have the Ronin root commit
+GECKO_RONIN_COMMIT=4b1114685cac9b758e923071eff5bf4350400ebd
+GAIA_RONIN_COMMIT=64cf0c8cf0cf2bf15f6fef5a8e6a1eeb40e18c95
+# Simple helper command to check that the current branch has a commit.
+git_has_commit = $(shell cd $(1) && git merge-base --is-ancestor $(2) HEAD && echo "yes" || echo "no")
```

Comment:
I might miss something, but couldn't you rely on the exit code only and test (in)equality with 0 where you use git_has_commit?